crates/turborepo-scm/src/git.rs

```diff
@@ -26,9 +32,23 @@ impl SCM {
         turbo_root: &AbsoluteSystemPath,
         from_commit: &str,
         to_commit: Option<&str>,
-    ) -> Result<HashSet<AnchoredSystemPathBuf>, Error> {
+        include_uncommitted: bool,
+        allow_unknown_objects: bool,
+    ) -> Result<ChangedFiles, Error> {
         match self {
-            Self::Git(git) => git.changed_files(turbo_root, from_commit, to_commit),
+            Self::Git(git) => {
+                match git.changed_files(turbo_root, from_commit, to_commit, include_uncommitted) {
+                    Ok(files) => Ok(ChangedFiles::Some(files)),
+                    Err(e) if allow_unknown_objects => {
```

Comment:
Can we only catch errors related to a commit not existing? I think ignoring all `git` errors might be a little heavy handed.

Reply:
It's a little tricky because the error for missing commit doesn't seem to have a single output. Like in the case where you have a shallow clone, you get:

```
fatal: main...HEAD: no merge base
```
but if you have a ref that just doesn't exist, you might get:
```
fatal: ambiguous argument 'main...nicholasyang/turbo-info': unknown revision or path not in the working tree.
```

but for an invalid commit hash, you'll get:

```
fatal: Invalid symmetric difference expression main...108774c7f7d999860668dcececd89d7e172ce20d
```

I'm not exactly sure which error message a user may get